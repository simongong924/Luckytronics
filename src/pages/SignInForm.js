import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class SignInForm extends Component {
    constructor() {
        super();

        this.state = {
            email: '',
            password: ''
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(e) {
        let target = e.target;
        let value = target.type === 'checkbox' ? target.checked : target.value;
        let name = target.name;

        this.setState({
          [name]: value
        });
    }
    handleClick() {
    this.setState(state => ({
      isToggleOn: !state.isToggleOn
    }));
  }

    handleSubmit(e) {
        e.preventDefault();
        this.props.history.push('/ticktform');

        console.log('The form was submitted with the following data:');
        console.log(this.state);
    }

    render() {
        return (
        <div className="FormCenter">
            <form onSubmit={this.handleSubmit} className="FormFields">
            <div className="FormField">
                <label className="FormField__Label" htmlFor="email">E-Mail Address</label>
                <input type="email" id="email" className="FormField__Input" placeholder="Enter your email" name="email" value={this.state.email} onChange={this.handleChange} />
              </div>

              <div className="FormField">
                <label className="FormField__Label" htmlFor="password">Password</label>
                <input type="password" id="password" className="FormField__Input" placeholder="Enter your password" name="password" value={this.state.password} onChange={this.handleChange} />
              </div>

              <div className="FormField">
                  <button className="FormField__Button mr-20">Sign In</button> <Link to="/sign-up" className="FormField__Link">Create new account</Link></div>
                  
            </form>
          </div>
        );
    }
}

export default SignInForm;