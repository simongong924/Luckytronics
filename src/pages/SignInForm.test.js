import React from 'react';
import { shallow } from 'enzyme';
import SignInForm from './SignInForm';
import renderer from 'react-test-renderer';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<SignInForm />',() => {
/*
it('has 2 inputs', () => {
const editor = shallow(<SignInForm />);
expect(editor.find('input').length).toEqual(2);
});
*/
it("snaps",() =>{
const component = renderer.create(
<SignInForm />
);
let tree = component.toJSON();

expect(tree).toMatchSnapshot();
});


});



/*
it('render correctly text component', () => {  
    const TextInputComponent = renderer.create(<SignInForm />).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});
*/
