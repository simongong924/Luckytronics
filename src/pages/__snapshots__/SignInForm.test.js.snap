// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> snaps 1`] = `
<div
  className="FormCenter"
>
  <h2>
     Sign In
  </h2>
  <form
    className="FormFields"
    onSubmit={[Function]}
  >
    <div
      className="FormField"
    >
      <label
        className="FormField__Label"
        htmlFor="email"
      >
        E-Mail Address
      </label>
      <input
        className="FormField__Input"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      className="FormField"
    >
      <label
        className="FormField__Label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="FormField__Input"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <div
      className="FormField"
    >
      <button
        className="FormField__Button mr-20"
      >
        Sign In
      </button>
       
    </div>
  </form>
</div>
`;
